lib/middleware/request-handler.ts

```diff
@@ -406,11 +412,17 @@ export function elevateRequest (req: Interface.IOurRequest,
         res.write(text + JSON.stringify(data));
     };
 
-    res.sendEnd = (status: string, message: string): void => {
-        var text = util.format('%s%s,%s}',
+    res.sendEnd = (status: string,
```

Comment:
Based on our IB discussion, here's the API I'd propose for this (I'll use the existing names although we may consider renaming them):

- `sendChunk` gets a boolean `isLast` parameter, to decouple the signaling of when chunk data is done from wrapping up the message.
- `sendEnd` adds the core properties (`status` and `message`) which are part of the overall API (i.e., every HTTP response we send has these properties which are used to signal error messages etc. and which I believe we should continue to do for all responses) and flushes the output.
- We introduce a new `sendOtherProperty` function (I know, that name sucks, but just for discussion) which writes a property to the output. This function either has to be constrained to be called before or after all `sendChunk` calls, or we can add buffering for when it's called in the middle and write them to the response during the last `sendChunk`.
- We also add a `sendWholeMessage` function, which may reuse `sendOtherProperty` and/or `sendEnd` internally.

Which leads to an API that says you have two choices:

1) Use `sendWholeMessage` and nothing else.
2) Use any number of `sendOther` and `sendChunk` calls, where the last `sendChunk` needs to specify that it is, followed by `sendEnd`.

Thoughts?